### PR TITLE
Add citation for random.orthogonal.

### PR DESCRIPTION
**Pull request by carlosgmartin**
https://github.com/google/jax/pull/23207#discussion_r1731137697

Rendered preview: https://jax--23257.org.readthedocs.build/en/23257/_autosummary/jax.random.orthogonal.html

### CODE REVIEW

**Comment by mattjj** [comment]
Just curious, why this reference and not e.g. one of the others from https://nhigham.com/2020/04/22/what-is-a-random-orthogonal-matrix/?

**Comment by carlosgmartin** [comment]
Anderson et al. (1987) and Stewart (1980) use different techniques. Mezzadri (2007) is this reference. Higham (2002) and Edelman and Rao (2005) seem to be more general references on the subjects of numerical algorithms and random matrix theory, respectively.

**Comment by carlosgmartin** [comment]
@dfm

File: jax/_src/random.py [hunk 1]
```diff
@@ -2030,6 +2030,8 @@ def orthogonal(
 ) -> Array:
   """Sample uniformly from the orthogonal group O(n).
 
+  Uses the technique described in https://arxiv.org/abs/math-ph/0609050.
```
Comment (jakevdp) [inline jax/_src/random.py:2033]: This should go further down. The documentation for complex behavior is arguably more relevant for most users than the arXiv link.

Maybe use a `References:` section?

Comment (jakevdp) [reply jax/_src/random.py:2033]: Example: https://github.com/google/jax/blob/8278a72b885f65546487284aed4494c098cafd07/jax/_src/scipy/linalg.py#L1806-L1813

Comment (carlosgmartin) [reply jax/_src/random.py]: @jakevdp Done.

**Review by jakevdp — APPROVED** [review]
Looks great, thanks!

**Pull request merged**